components/script/dom/htmlformelement.rs

```diff
@@ -417,55 +411,85 @@ impl HTMLFormElement {
 
         let mut load_data = LoadData::new(action_components, doc.get_referrer_policy(), Some(doc.url().clone()));
 
-        let parsed_data = match enctype {
-            FormEncType::UrlEncoded => {
-                load_data.headers.set(ContentType::form_url_encoded());
-
-                form_urlencoded::Serializer::new(String::new())
-                    .encoding_override(Some(self.pick_encoding()))
-                    .extend_pairs(form_data.into_iter().map(|field| (field.name.clone(), field.value_str())))
-                    .finish()
-            }
-            FormEncType::FormDataEncoded => {
-                let boundary = self.generate_boundary();
-                let mime = mime!(Multipart / FormData; Boundary =(&boundary));
-                load_data.headers.set(ContentType(mime));
-
-                self.encode_multipart_form_data(&mut form_data, None, boundary)
-            }
-            FormEncType::TextPlainEncoded => {
-                load_data.headers.set(ContentType(mime!(Text / Plain)));
-
-                self.encode_plaintext(&mut form_data)
-            }
-        };
-
         // Step 18
         let win = window_from_node(self);
```

Comment:
Actually, `plan_to_navigate` doesn't even need to have a `&Window` argument at all, since it can just use `window_from_node` in its own function body.